src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -139,56 +125,31 @@ public void removePerson(Person key) {
      * @param sg the SuperGroup to be added into AddressBook.
      */
     public void addSuperGroup(SuperGroup sg) {
-        if (!superGroups.containsKey(sg.getName())) {
-            superGroups.put(sg.getName(), sg);
-        }
+        superGroups.add(sg);
     }
```

Comment:
This is what I was talking about in point 2 on the PR thread, We could simply have this be a call to superGroups.add command, and have the add command there do all the handling of renaming or checking, since we already built the UniqueList data structure for this kind of operation.

Reply:
So I suppose this is all good lol